File: back/src/main/java/com/nrabello/back/domain/Dimension.java
package com.nrabello.back.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

@Entity
@Table(name = "Dimensoes")
public class Dimension {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="dim_id")
    private Long id;
    @Column(name="dim_altura")
    private Double height;

    @Column(name="dim_largura")
    private Double width;

    @Column(name="dim_profundidade")
    private Double depth;

    @Column(name="dim_peso")
    private Double weight;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name="dim_liv_id", referencedColumnName = "liv_id")
    @JsonIgnore
    private Book book;

    public Dimension() {
    }

    public Dimension(Long id, Double height, Double width, Double depth, Double weight, Book book) {
        this.setId(id);
        this.setHeight(height);
        this.setWidth(width);
        this.setDepth(depth);
        this.setWeight(weight);
        this.setBook(book);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Double getHeight() {
        return height;
    }

    public void setHeight(Double height) {
        this.height = height;
    }

    public Double getWidth() {
        return width;
    }

    public void setWidth(Double width) {
        this.width = width;
    }

    public Double getDepth() {
        return depth;
    }

    public void setDepth(Double depth) {
        this.depth = depth;
    }

    public Double getWeight() {
        return weight;
    }

    public void setWeight(Double weight) {
        this.weight = weight;
    }

    public Book getBook() {
        return book;
    }
    public void setBook(Book book) {
        this.book = book;
    }
}

File: front/src/models/Dimension.ts

export interface IDimensionModel {
    id?: number;
    height?: number;
    width?: number;
    depth?: number;
    weight?: number;
  }
  
  export class Dimension {
    id: number;
    height: number;
    width: number;
    depth: number;
    weight: number;
    constructor({
        id = 0,
        height = 0,
        width = 0,
        depth = 0,
        weight = 0,
    }: IDimensionModel = {}) {
        this.id = id;
        this.height = height;
        this.width = width;
        this.depth = depth;
        this.weight = weight;
    }

    // get id(): number {
    //   return this._id;
    // }
    // set id(id: number) {
    //   this._id = id;
    // }
    
    // get height(): number {
    //   return this._height;
    // }
    // set height(height: number) {
    //   this._height = height;
    // }

    // get width(): number {
    //   return this._width;
    // }
    // set width(width: number) {
    //   this._width = width;
    // }

    // get depth(): number {
    //   return this._depth;
    // }
    // set depth(depth: number) {
    //   this._depth = depth;
    // }

    // get weight(): number {
    //   return this._weight;
    // }
    // set weight(weight: number) {
    //   this._weight = weight;
    // }

}